### PR TITLE
vimPlugins: update

### PR DESCRIPTION
**Pull request by KAction**
- vimPlugins: add new plugins
- vimPlugins: update

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
Each plugin addition should probably be their own commit.

When adding new vim plugins, please follow [directions for adding new vim plugins](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs).

The steps should roughly be:
```
git checkout -b <branch-name>
GITHUB_API_TOKEN=[personal token] ./pkgs/misc/vim-plugins/update.py --add "[owner]/[name]"
git push <fork> <branch>
```
you should see the following in your commit history:
```
vimPlugins: update
vimPlugins.[name]: init at <date>
```
If the package needs additional patching or you want to include a meta section, please put the related changes in `pkgs/misc/vim-plugins/overrides.nix`. Do not edit the generated.nix, as those changes will be lost next update.

Why GITHUB_API_TOKEN?
Github limits github queries to 100/hr if you're a guest, but if you have a personal token, it will raise it to 5000/hr. The vim plugin update script will query the repositories to see what the latest release was, which will exceed the unauthenticated query limit.

For documentation on creating a token, please refer [to the official docs](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token#creating-a-token)

File: pkgs/misc/vim-plugins/vim-plugin-names [hunk 1]
```diff
@@ -855,6 +857,7 @@ vijaymarupudi/nvim-fzf-commands
 vim-airline/vim-airline
 vim-airline/vim-airline-themes
 vim-autoformat/vim-autoformat
+vim-autoformat/vim-autoformat
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/vim-plugin-names:860]: This already exists

**Review by jonringer — CHANGES_REQUESTED** [review]

File: pkgs/misc/vim-plugins/generated.nix [hunk 2]
```diff
@@ -7013,6 +7025,18 @@ final: prev:
...
+    version = "2021-11-07";
+    src = fetchFromGitHub {
+      owner = "vim-autoformat";
+      repo = "vim-autoformat";
+      rev = "bb11f30377985e45e2eecef570856d42dbabb8b0";
+      sha256 = "1lx5lrb0hjijvwngvgsgchww70c1bgqf5qxj8lhvk11dvxn3k087";
+    };
+    meta.homepage = "https://github.com/vim-autoformat/vim-autoformat/";
+  };
+
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/generated.nix:7039]: this needs to be removed, as it's already defined

**Review by jonringer — CHANGES_REQUESTED** [review]

File: pkgs/misc/vim-plugins/generated.nix (hunk 2 above)
Comment (KAction) [reply pkgs/misc/vim-plugins/generated.nix]: My bad. Forgot that update.py does not filter-out duplicates. Fixed

**Pull request closed without merging**